schemas-extractor/template/generate-schema.go

```diff
@@ -190,13 +194,39 @@ func DoGenerate(provider *schema.Provider, providerName string, outputFilePath s
 
 type SchemaElement struct {
 	// One of "schema.ValueType" or "SchemaElements" or "SchemaInfo"
-	Type string `json:",omitempty"`
+	Type string
 	// Set for simple types (from ValueType)
-	Value string `json:",omitempty"`
+	Value *string
 	// Set if Type == "SchemaElements"
-	ElementsType string `json:",omitempty"`
+	ElementsType string
 	// Set if Type == "SchemaInfo"
-	Info SchemaInfo `json:",omitempty"`
+	Info SchemaInfo
+}
+
+func (e *SchemaElement) MarshalJSON() ([]byte, error) {
+	if e.Value == nil {
```

Comment:
Maybe there is a better way?  I don't know much about Go.

Reply:
Neither do I :)

Reply:
Origin
```
Value string `json:",omitempty"`
```
Should do the trick. You don't need to specify separate Marshaller for this.

And `string` in go is actually `[]byte` which is actually already pointer type. And in most cases it is better to use just `string` instead of `*string`. And compare `string` with empty string (`""`) instead of null. What is literally the same.

I don't understand why original `omitempty` doesn't work.